monitoring/base/rhods-rules.yaml

```diff
@@ -16,5 +16,9 @@ spec:
       expr: count(kube_pod_container_status_ready{namespace="rhods-notebooks",pod=~"jupyterhub-nb.*"}==1)
     - record: cluster:usage:consumption:rhods:active_users
       expr: count(count_over_time(kube_pod_container_status_ready{namespace="rhods-notebooks",pod=~"jupyterhub-nb.*"}[1h]))
+    - record: cluster:usage:consumption:cpu:requests:runtime
+        expr: sum(kube_pod_container_resource_requests{namespace='rhods-notebooks', resource='cpu'})
```

Comment:
You've added these metrics in the right place, but the calculations/expressions themselves won't give us exactly what we're looking for.

What you have so far just gives us the sum of all the CPU requests (and limits in the next case) for all RHODS notebook pods. What we want is the sum of pod runtime * pod CPU requests (or limits) for all RHODS notebook pods. Example: if a notebook pod runs for 1 hour and it has a request for 4 CPUs, the value of our target metric should be 4 after one hour. If we have 2 pods each with 4 CPUs, the value should be 8 after 1 hour `(1*4) + (1*4)`

Reply:
Expanding on this a bit:

The expression for the above metric `cluster:usage:consumption:rhods:pod:up` gives you the sum of notebook pod runtime. I think you could take the product of the two underlying metrics and sum the result to get what we want. This may not be exactly right, but is where I'd start.

Reply:
After digging in on this more with Dharmit, I'm wrong. Resolving.

Reply:
Updated queries to include sum over time as per our discussion @accorvin .